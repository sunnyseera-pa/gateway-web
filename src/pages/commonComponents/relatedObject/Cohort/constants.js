const tab_url = '/search?search=&tab=Cohorts';
export const cohort = {
	TAB: 'Cohort',
	FC: {
		parentKey: 'cohortinclusionexclusion',
		filter: 'cohortFilterCriteriaSelected',
		tagType: 'tag',
		url: `${tab_url}&cohortinclusionexclusion=`,
	},
};
