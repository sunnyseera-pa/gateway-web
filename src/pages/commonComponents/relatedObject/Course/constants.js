const tab_url = '/search?search=&tab=Courses';
export const course = {
	TAB: 'Course',
	AWARDS: {
		parentKey: 'courseaward',
		filter: 'courseaward',
		tagType: 'tag',
		url: `${tab_url}&courseaward=`,
	},
	DOMAINS: {
		parentKey: 'coursedomains',
		filter: 'courseDomainsSelected',
		tagType: 'tag',
		url: `${tab_url}&coursedomains=`,
	},
};
