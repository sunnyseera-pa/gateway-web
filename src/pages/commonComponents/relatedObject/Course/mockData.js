export default {
    _id: '5f96f3450d236c6cafdaa473',
    domains: ['test'],
    award: ['BSc'],
    courseOptions: [
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
    ],
    id: 3491286126459112,
    type: 'course',
    activeflag: 'active',
    title: 'Test A course',
    provider: 'Test A',
    description: 'Test A Test A Test A edited Test A',
    counter: 208,
    relatedresources: 3,
    updated: 1621465200000,
};
