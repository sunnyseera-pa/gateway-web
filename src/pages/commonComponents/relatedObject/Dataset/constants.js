const tab_url = '/search?search=&tab=Datasets';
export const dataset = {
	TAB: 'Dataset',
	PHENOTYPES: {
		parentKey: 'phenotypes',
		filter: 'phenotypes',
		tagType: 'phenotype',
		url: `${tab_url}&phenotypes=`,
	},
	FEATURES: {
		parentKey: 'datasetfeatures',
		filter: 'datasetfeatures',
		tagType: 'tag',
		url: `${tab_url}&datasetfeatures=`,
	},
};
