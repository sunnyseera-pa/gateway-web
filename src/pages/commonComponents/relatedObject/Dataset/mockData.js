export default {
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['COVID-19', 'TRACKER', 'SAIL', 'COVID', 'MOBILE APP', 'CORONAVIRUS', 'SYMPTOM', 'NCS', 'National Core Study'],
        topics: [],
    },
    datasetfields: {
        publisher: 'ALLIANCE > TestSAIL',
        geographicCoverage: ['Great Britain'],
        physicalSampleAvailability: ['NOT AVAILABLE'],
        abstract: 'This is a test dataset abstract',
        ageBand: '15-150',
        metadataquality: {
            quality_score: 89.48,
        },
        phenotypes: [],
    },
    programmingLanguage: [],
    pid: 'c1f4b16c-9dfa-48e5-94ee-f0aa58c270e4',
    id: 401490453939233,
    datasetid: '78ce070c-84db-4676-81c5-0c672f4e12e3',
    type: 'dataset',
    activeflag: 'active',
    name: 'COVID-19 Symptom Tracker Dataset',
    description: 'This is a test dataset description',
    license: 'https://covid.joinzoe.com/privacy-notice',
    datasetv2: {
        summary: {
            publisher: {
                name: 'sail',
                logo: '',
                memberOf: 'ALLIANCE',
            },
        },
    },
    counter: 955,
    isCohortDiscovery: false,
    persons: [
        {
            firstname: 'Rich',
            lastname: 'Example',
        },
    ],
    latestUpdate: '2021-05-04T14:14:49.389Z',
    relatedresources: 1,
};
