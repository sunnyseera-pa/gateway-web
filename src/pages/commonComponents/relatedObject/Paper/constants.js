const tab_url = '/search?search=&tab=Paper';
export const paper = {
    TAB: 'Paper',
    FEATURES: {
        parentKey: 'paperfeatures',
        filter: 'paperFeaturesSelected',
        tagType: 'tag',
        url: `${tab_url}&paperfeatures=`,
    },
    TOPICS: {
        parentKey: 'papertopics',
        filter: 'paperTopicsSelected',
        tagType: 'tag',
        url: `${tab_url}&papertopics=`,
    },
};
