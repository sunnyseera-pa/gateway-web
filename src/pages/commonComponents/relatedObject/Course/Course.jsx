/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import { Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { has, isEmpty, isString, isObject } from 'lodash';
import moment from 'moment';
import { ReactComponent as CalendarSvg } from '../../../../images/calendaricon.svg';
import { stripMarkdown } from '../../../../utils/GeneralHelper.util';
import SVGIcon from '../../../../images/SVGIcon';
import RemoveButton from '../RemoveButton/RemoveButton';
import Title from '../Title/Title';
import Description from '../Description/Description';
import Tag from '../Tag/Tag';
import { course } from './constants';
import '../../CommonComponents.scss';

const Course = ({ data, activeLink, onSearchPage, showRelationshipQuestion, updateOnFilterBadge, removeButton, onClick }) => {
    const renderCourseDetails = () => {
        const courseRender = [];
        const { courseOptions } = data;
        has(courseOptions[0], 'startDate') && isObject(courseOptions[0].startDate)
            ? courseRender.push(
                  <span data-testid='course-start-date'> Starts {moment(courseOptions.startDate).format('dddd Do MMMM YYYY')} </span>
              )
            : courseRender.push(<span data-testid='course-start-date'> Flexible dates </span>);

        has(courseOptions[0], 'studyMode') &&
            isString(courseOptions[0].studyMode) &&
            courseRender.push(<span data-testid='course-study-mode'> | {courseOptions[0].studyMode} </span>);

        !onSearchPage &&
            !isEmpty(courseOptions[1]) &&
            courseOptions.map((courseOption, index) =>
                courseRender.push(<>{index > 0 && <span data-testid='course-study-mode-extra'> ,{courseOption.studyMode} </span>}</>)
            );

        return courseRender;
    };
    return (
        <Row data-testid='related-course-object' className='noMargin'>
            <Col sm={10} lg={10} className='pad-left-24'>
                <Title
                    activeLink={activeLink}
                    name={data.title}
                    id={data.id}
                    type={data.type}
                    onClickHandler={() => {
                        onClick();
                    }}
                />
                <br />
                <Title
                    activeLink={false}
                    name={data.provider}
                    id={data.provider}
                    type={data.type}
                    className={activeLink ? 'gray800-14 underlined' : 'gray800-14'}
                    onClickHandler={() => updateOnFilterBadge('courseProviderSelected', data.provider)}
                />

                <Row className='margin-top-8'>
                    <Col sm={12} lg={12}>
                        <CalendarSvg className='calendarSVG' />
                        <span className='gray800-14 margin-left-10'>{renderCourseDetails()}</span>
                    </Col>
                </Row>
            </Col>
            <Col sm={2} lg={2} className='pad-right-24'>
                {showRelationshipQuestion && <RemoveButton removeButtonHandler={removeButton} />}
            </Col>
            <Col sm={12} lg={12} className='pad-left-24 pad-right-24 pad-top-16'>
                <Tag tagName={course.TAB} tagType={data.type} updateOnFilterBadgeHandler={updateOnFilterBadge}>
                    <SVGIcon name='educationicon' fill='#ffffff' className='badgeSvg mr-2' viewBox='-2 -2 22 22' />
                </Tag>

                {data.award &&
                    data.award.map((award, index) => (
                        <Tag
                            key={`${award}-${index}`}
                            tagName={award}
                            activeLink={activeLink}
                            onSearchPage={onSearchPage}
                            updateOnFilterBadgeHandler={updateOnFilterBadge}
                            {...course.AWARDS}
                        />
                    ))}

                {data.domains &&
                    data.domains.map((domain, index) => (
                        <Tag
                            key={`${domain}-${index}`}
                            tagName={domain}
                            activeLink={activeLink}
                            onSearchPage={onSearchPage}
                            updateOnFilterBadgeHandler={updateOnFilterBadge}
                            {...course.DOMAINS}
                        />
                    ))}
            </Col>
            {!showRelationshipQuestion && <Description type={data.type} description={stripMarkdown(data.description, 255)} />}
        </Row>
    );
};

Course.propTypes = {
    data: PropTypes.object.isRequired,
    activeLink: PropTypes.bool.isRequired,
    showRelationshipQuestion: PropTypes.bool.isRequired,
    onSearchPage: PropTypes.bool.isRequired,
    updateOnFilterBadge: PropTypes.func.isRequired,
    removeButton: PropTypes.func.isRequired,
};

Course.defaultProps = {
    onClick: () => {},
};

export default Course;
