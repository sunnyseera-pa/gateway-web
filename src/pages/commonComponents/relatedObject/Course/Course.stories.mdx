import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Course from './Course';
import mockData from './mockData';

<Meta
	title='Related Resources/Course'
	component={Course}
	args={{
		data: mockData,
		onSearchPage: false,
		activeLink: false,
		showRelationshipQuestion: false,
	}}
/>

<Canvas>
	<Story name='Course'>
		{props => {
			const updateOnFilterBadge = () => {
				alert('updateOnFilterBadge');
			};
			const removeButton = () => {
				alert('removed');
			};
			return <Course {...props} updateOnFilterBadge={updateOnFilterBadge} removeButton={removeButton} />;
		}}
	</Story>
</Canvas>
