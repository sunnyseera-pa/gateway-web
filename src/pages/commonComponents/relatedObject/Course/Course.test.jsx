import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Course from './Course';
import { course } from './constants';
import mockData from './mockData';
const props = {
	data: { ...mockData },
	onSearchPage: false,
	activeLink: false,
	showRelationshipQuestion: false,
	updateOnFilterBadge: jest.fn(),
	removeButton: jest.fn(),
};
let wrapper;

describe('Given the Course component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<Course {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then Course Title should be rendered with Provider', () => {
			expect(screen.getByTestId(`title-${props.data.type}-${props.data.id}`)).toHaveTextContent(props.data.title);
			expect(screen.getByTestId(`title-${props.data.type}-${props.data.provider}`)).toHaveTextContent(props.data.provider);
		});

		it('Then Course SVG Icon should be rendered', () => {
			expect(screen.getByTestId('educationicon')).toBeTruthy();
		});

		it('Then the Awards Badge  should be rendered without links', () => {
			props.data.award.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
			});
		});

		it('Then the Domain Badges should be rendered without links', () => {
			props.data.domains.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
			});
		});

		it('Then the course date and study mode should be rendered', () => {
			expect(screen.getByTestId('course-start-date')).toHaveTextContent('Flexible dates');
			expect(screen.getByTestId(`course-study-mode`)).toHaveTextContent(`| ${props.data.courseOptions[0].studyMode}`);
			expect(screen.getByTestId(`course-study-mode-extra`)).toHaveTextContent(`,${props.data.courseOptions[1].studyMode}`);
		});
	});

	describe('And activeLink is true', () => {
		it('Then the Tilte should be clickable with a link', () => {
			const { rerender } = wrapper;
			rerender(<Course {...props} activeLink={true} />);
			expect(screen.getByTestId(`title-${props.data.type}-${props.data.id}`)).toHaveAttribute('href', `/course/${props.data.id}`);
		});
		it('Then the Awards Badge/Tag should be rendered with links', () => {
			props.data.award.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.getByTestId(`badge-${value}-link`)).toHaveAttribute('href', `${course.AWARDS.url}${value}`);
			});
		});

		it('Then the Domains Badge/Tag should be rendered with links', () => {
			props.data.domains.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.getByTestId(`badge-${value}-link`)).toHaveAttribute('href', `${course.DOMAINS.url}${value}`);
			});
		});

		describe('And onSearchPage is true', () => {
			let updateOnFilterBadge = jest.fn();
			it('Then Badge Awards should be rendered without links', () => {
				const { rerender } = wrapper;
				rerender(<Course {...props} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />);
				props.data.award.map(value => {
					expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
					expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
				});
			});
			it('Then the Awards badge updateOnFilterBadge should be called', () => {
				fireEvent.click(screen.getByTestId(`badge-${props.data.award[0]}`));
				expect(updateOnFilterBadge.mock.calls.length).toBe(1);
				expect(updateOnFilterBadge.mock.calls[0][0]).toEqual(course.AWARDS.filter);
				expect(updateOnFilterBadge.mock.calls[0][1]).toEqual({ label: props.data.award[0], parentKey: course.AWARDS.parentKey });
			});

			it('Then the Domain Badge/Tag updateOnFilterBadge should be called', () => {
				const { rerender } = wrapper;
				let updateOnFilterBadgeTopic = jest.fn();
				rerender(<Course {...props} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadgeTopic} />);
				fireEvent.click(screen.getByTestId(`badge-${props.data.domains[0]}`));
				expect(updateOnFilterBadgeTopic.mock.calls.length).toBe(1);
				expect(updateOnFilterBadgeTopic.mock.calls[0][0]).toEqual(course.DOMAINS.filter);
				expect(updateOnFilterBadgeTopic.mock.calls[0][1]).toEqual({ label: props.data.domains[0], parentKey: course.DOMAINS.parentKey });
			});

			it('Then the course date and study mode should be rendered with no extra', () => {
				expect(screen.getByTestId('course-start-date')).toHaveTextContent('Flexible dates');
				expect(screen.getByTestId(`course-study-mode`)).toHaveTextContent(`| ${props.data.courseOptions[0].studyMode}`);
				expect(screen.queryByTestId(`course-study-mode-extra`)).toBeNull();
			});
		});
	});
	describe('And showRelationshipQuestion is true', () => {
		it('Then the remove button should be rendered ', () => {
			const { rerender } = wrapper;
			rerender(<Course {...props} showRelationshipQuestion={true} />);
			expect(screen.getByTestId('closeicon')).toBeTruthy();
		});
		it('Then onclick removeButton function should be called', () => {
			fireEvent.click(screen.getByTestId('closeicon'));
			expect(props.removeButton.mock.calls.length).toBe(1);
		});
		it('Then the description should not be rendered', () => {
			expect(screen.queryByTestId('paper-description')).toBeNull();
		});
	});
});
