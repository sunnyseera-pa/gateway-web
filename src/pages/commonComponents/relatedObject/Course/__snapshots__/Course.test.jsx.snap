// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Course component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="noMargin row"
    data-testid="related-course-object"
  >
    <div
      class="pad-left-24 col-lg-10 col-sm-10"
    >
      <span
        class="black-bold-16"
        data-testid="title-course-3491286126459112"
      >
         
        Test A course
      </span>
      <br />
      <span
        class="gray800-14"
        data-testid="title-course-Test A"
      >
         
        Test A
      </span>
      <div
        class="margin-top-8 row"
      >
        <div
          class="col-lg-12 col-sm-12"
        >
          <iconmock
            class="calendarSVG"
          />
          <span
            class="gray800-14 margin-left-10"
          >
            <span
              data-testid="course-start-date"
            >
               Flexible dates 
            </span>
            <span
              data-testid="course-study-mode"
            >
               | 
              Full-time
               
            </span>
            <span
              data-testid="course-study-mode-extra"
            >
               ,
              Full-time
               
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="pad-right-24 col-lg-2 col-sm-2"
    />
    <div
      class="pad-left-24 pad-right-24 pad-top-16 col-lg-12 col-sm-12"
    >
      <div
        class="badge-course"
        data-testid="badge-Course"
      >
        <svg
          class="badgeSvg mr-2"
          data-testid="educationicon"
          height="100%"
          viewBox="-2 -2 22 22"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M12.168.142c.534-.192 1.12-.19 1.651.01l10.506 3.914c1.243.463 1.874 1.845 1.411 3.087-.281.755-.923 1.319-1.708 1.5l-2.527.584v3.293c0 1.071-.709 2.005-1.724 2.303l-.163.041-1.614.353v.809l.731.857c1.211 1.457 1.73 2.564 1.173 3.84l-.079.168c-1.006 1.996-4.6 1.958-5.577-.013-.63-1.272-.205-2.318.945-3.775l.317-.39.49-.593v-.465l-2.495.547c-.272.06-.553.071-.828.035l-.205-.036-6.016-1.335c-1.098-.243-1.88-1.218-1.88-2.343V9.224L1.82 8.54C.59 8.233-.182 7.025.037 5.792l.034-.16c.194-.778.762-1.408 1.516-1.68zM17.048 18c-.834 1.006-1.17 1.672-1.008 2 .243.491 1.734.527 2 0 .177-.352-.153-1.018-.992-2zm-3.99-11.765c-2.128 0-4.285.82-6.481 2.494l-.001 3.804c0 .188.13.35.313.39l6.016 1.335c.057.013.115.013.172 0l2.923-.64V6.755c-.987-.348-1.968-.52-2.942-.52zM18 7.692v5.488l1.186-.26c.158-.034.276-.158.307-.311l.008-.079V8.7c-.503-.38-1.003-.716-1.5-1.008zm0-3.849v1.584c.996.482 1.985 1.109 2.968 1.88l2.61-.602c.13-.03.237-.125.284-.25.077-.207-.028-.438-.235-.515L18 3.843zm-4.88-1.818c-.088-.033-.185-.033-.274-.001L2.265 5.834c-.126.045-.22.15-.253.28-.053.214.078.431.292.485l2.845.706c2.597-2.035 5.238-3.07 7.909-3.07.985 0 1.966.14 2.942.42V3.099z"
            fill="#ffffff"
          />
        </svg>
        <span>
          Course
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-BSc"
      >
        <span>
          BSc
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-test"
      >
        <span>
          test
        </span>
        <span />
      </div>
    </div>
    <div
      class="pad-left-24 pad-right-24 pad-top-24 pad-bottom-16 col-lg-12 col-sm-12"
    >
      <span
        class="gray800-14"
        data-testid="course-description"
      >
        Test A Test A Test A edited Test A
      </span>
    </div>
  </div>
</div>
`;
